src/Collisions/collider.ts

```diff
@@ -138,7 +140,14 @@ export class Collider {
     /** @hidden */
     public _initialize(source: Vector3, dir: Vector3, e: number): void {
         this._velocity = dir;
-        Vector3.NormalizeToRef(dir, this._normalizedVelocity);
+        this._velocitySquaredLength = this._velocity.lengthSquared();
+        const len = Math.sqrt(this._velocitySquaredLength)
+        if (len === 0 || len === 1.0) {
+            this._normalizedVelocity.copyFromFloats(dir._x, dir._y, dir._z);

```

Comment:
could use copyFrom instead of using the 3 floats

Reply:
Yes, sure

Reply:
`copyFrom` call `copyFromFloats`, so directly using `copyFromFloats` saves a call ;)

However, we should probably change the implementation of `Vector3.copyFrom` to directly store the values instead of calling `Vector3.copyFromFloats`, as it is what `Vector2.copyFrom`, `Vector4.copyFrom` and `Quaternion.copyFrom` are doing...

Reply:
Yes, that would be wonderful. The reason I commented is that when I see `copyFrom` it is obvious what type this input object is, but copy from float might mean that even if it IS a vector something is manipulated between input and output. 
Internally it does the same, of course. We can change it - @Popov72 want to submit this?